vm/src/frame.rs

```diff
@@ -538,11 +538,14 @@ impl ExecutingFrame<'_> {
             }
             bytecode::Instruction::LoadNameAny(idx) => {
                 let name = self.code.names[*idx as usize];
-                let value = self.locals.mapping().subscript(name, vm).ok();
-                self.push_value(match value {
-                    Some(x) => x,
-                    None => self.load_global_or_builtin(name, vm)?,
-                });
+                let result = self.locals.mapping().subscript(name, vm);
+                match result {
+                    Ok(x) => self.push_value(x),
+                    Err(e) if e.class().is(vm.ctx.exceptions.key_error) => {
```

Comment:
Tiny issue, we also need to also accept subclasses of `KeyError`:

```suggestion
                    Err(e) if e.fast_isinstance(vm.ctx.exceptions.key_error) => {
```

Reply:
Code fix completed.